jetbrains-core/src/software/aws/toolkits/jetbrains/core/notification/NoticeType.kt

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.notification
+
+import com.intellij.openapi.application.ApplicationInfo
+import com.intellij.openapi.application.ApplicationNamesInfo
+import com.intellij.openapi.extensions.ExtensionPointName
+import software.aws.toolkits.resources.message
+
+interface NoticeType {
```

Comment:
Should we allow for additional actions? For instance, for the example case, you may want a link to upgrade the IDE.

Reply:
That's a good thought. I would consider this out of scope for this change, but `NoticeType` could certainly expose something like `getActions` which the Manager could add to the notification that is produced.

Reply:
I'm on the fence about considering this out-of-scope for the change, since the task is about writing the framework for these changes as well.

Reply:
YAGNI - it'll be easy to add in a backwards compatible way in the future - I tend to agree with @awschristou that it's not a requirement for this change. We've done as much of the generic version as we needed for our requirement and left it open for other notices to extend in the future.